### PR TITLE
Upgrade OptimismMintableERC20Factory implementation

### PR DESCRIPTION
**Pull request by xenoliss**
This PR add the `RunUpgradeToAndCallERC20Factory` to upgrade the `OptimismMintableERC20Factory` implementation, at address [0x4200000000000000000000000000000000000012](https://basescan.org/address/0x4200000000000000000000000000000000000012#code), to point to the newly deployed factory at address [0x6922ac4DbDfEdEa3a1E5535f12c3171f2b964C91](https://basescan.org/address/0x6922ac4dbdfedea3a1e5535f12c3171f2b964c91#code), effectively replacing the old implementation at address [0xc0D3c0d3C0d3c0d3c0D3c0d3c0D3c0D3c0D30012](https://basescan.org/address/0xc0d3c0d3c0d3c0d3c0d3c0d3c0d3c0d3c0d30012#code).

### CODE REVIEW

File: mainnet/2024-04-17-upgrade-erc20-factory/README.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+``` shell
+make sign-op # or make sign-cb for Coinbase signers
+```
+
+Note: there have been reports of some folks seeing this error `Error creating signer: error opening ledger: hidapi: failed to open device`. A fix is in progress, but not yet merged. If you come across this, open a new terminal and run the following to resolve the issue:
+```
+git clone git@github.com:base-org/eip712sign.git
+cd eip712sign
+go install
+```
```
Comment (stevieraykatz) [inline mainnet/2024-04-17-upgrade-erc20-factory/README.md:39]: This is no longer relevant with the updated eip712 signer go lib.

File: mainnet/2024-04-17-upgrade-erc20-factory/README.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+3. Validate and extract domain hash and message hash to approve.
+4. Validate that the transaction completed successfully
+
+
+#### 3.1. Validate integrity of the simulation.
+
+Make sure you are on the "Overview" tab of the tenderly simulation, to
+validate integrity of the simulation, we need to check the following:
+
+1. "Network": Check the network is Optimism Mainnet.
```
Comment (stevieraykatz) [inline mainnet/2024-04-17-upgrade-erc20-factory/README.md:62]:
```suggestion
1. "Network": Check the network is Base Mainnet.
```

File: mainnet/2024-04-17-upgrade-erc20-factory/foundry.toml [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+solc_version = "0.8.19"
+via-ir = true
+remappings = [
+    '@eth-optimism-bedrock/=lib/optimism/packages/contracts-bedrock/',
+    '@openzeppelin/contracts/=lib/openzeppelin-contracts/contracts',
+    '@openzeppelin/contracts-upgradeable/=lib/openzeppelin-contracts-upgradeable/contracts',
+    '@rari-capital/solmate/=lib/solmate/',
+    '@base-contracts/=lib/base-contracts',
+    'solady/=lib/solady/src/',
+    '@agora=lib/optimism-gov/src',
```
Comment (stevieraykatz) [inline mainnet/2024-04-17-upgrade-erc20-factory/foundry.toml:18]: can remove

File: mainnet/2024-04-17-upgrade-erc20-factory/script/RunUpgradeToAndCallERC20Factory.sol [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+contract RunUpgradeToAndCallERC20Factory is NestedMultisigBuilder {
+    address public BRIDGE = vm.envAddress("BRIDGE");
+    address public PROXY_ADMIN = vm.envAddress("PROXY_ADMIN");
+    address public ERC20_FACTORY = vm.envAddress("ERC20_FACTORY");
+    address public ERC20_FACTORY_IMPL = vm.envAddress("ERC20_FACTORY_IMPL");
+
+    address internal NESTED_SAFE = vm.envAddress("NESTED_SAFE");
+    address internal BASE_SAFE = vm.envAddress("BASE_SAFE");
+    address internal OP_SAFE = vm.envAddress("OP_SAFE");
```
Comment (stevieraykatz) [inline mainnet/2024-04-17-upgrade-erc20-factory/script/RunUpgradeToAndCallERC20Factory.sol:25]: These aren't needed in the script's context and can be safely removed.

**Review by wilsoncusack — APPROVED** [review]

**Pull request merged**